server/src/main/java/com/broadcom/lsp/cobol/core/messages/LogLevel.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.messages;
+
+import com.google.common.collect.Iterables;
+import com.google.gson.JsonPrimitive;
+import lombok.experimental.UtilityClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Update the log level for the application.
+ *
+ * <p>IMPORTANT: This class relies on the concrete implementations by logback. SLF4J don't expose
+ * any contract to update the log level on runtime.
+ *
+ * <p>In case of wrong input, log level defaults to {@link ch.qos.logback.classic.Level#ERROR}
+ */
+@UtilityClass
+public class LogLevel {
+
+  private static final ch.qos.logback.classic.Logger rootLogger =
+      (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+
+  /**
+   * Update loglevel for the entire application.
+   *
+   * @return Consumer#List#Object , currently supports List#String and List#JsonPrimitive
+   */
+  public static Consumer<List<Object>> updateLogLevel() {
+    return loglevel -> {
+      if (Objects.nonNull(loglevel)) {
+        Object localeNode = Iterables.getFirst(loglevel, "");
+        if (localeNode instanceof JsonPrimitive) {
```

Comment:
Below you provide: logLevelNode.getAsString() which returns String, then why you need to check for "instanceof JsonPrimitive"

Reply:
getAsString() is a method on JsonPrimitive, that's why casting it to JsonPrimitive upon instanceof check.

`private static void updateLogLevel(JsonPrimitive logLevelNode)`

accepts JsonPrimitive